### PR TITLE
Add httpService and authService to the frontend

### PR DESCRIPTION
**Pull request by vitalii0o**
> Add auth store module with actions, getters, etc.
> 
> const getters = {
>   isLoggedIn,
>   getAuthenticatedUser,
>   getToken
> }
> 
> const mutations {
>   USER_LOGOUT,
>   SET_TOKEN, 
>   SET_AUTHENTICATED_USER
>   }

1. axios package https://github.com/axios/axios
2. action login
3. action logout
4. action resetPassword
5. action refreshToken
6. action sendForgotEmail
7. action fetchAuthenticatedUser

### CODE REVIEW

File: frontend/src/service/httpService.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import axios from 'axios';
+import store from '../store';
+import authCookie from './auth';
+import configuration from '../config';
+import state from '../state';
+
+const httpService = axios.create({
```
Comment (lenchvolodymyr) [inline frontend/src/service/httpService.js]: You created a little incorrectly, I think service has to contain two methods request() and authRequest(), in order to you can use it like this

```
httpService.authRequest({
                url: configuration.LogoutUri,
                method: 'post',
                data: {
                    user
                }
            }).
```

So it is well ` axios.create({`, but store it in another variable and use it inside methods

File: frontend/src/services/common/httpService.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    put(url, data) {
+        return this.axios.put(url, data);
+    }
+
+    delete(url, params) {
+        return this.axios.delete(url, params);
+    }
+
+    static authRequest() {
```
Comment (nemoipaha) [inline frontend/src/services/common/httpService.js]: No need for this now )

File: frontend/src/services/common/httpService.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+export class HttpService {
+    constructor() {
+        this.axios = axios;
+
+        this.authAxios = axios.create({
+            baseURL: config.host,
+            //timeout: config.TIMEOUT || 10000
+        });
+
+        this.authAxios.interceptors.request.use(config => {
```
Comment (nemoipaha) [inline frontend/src/services/common/httpService.js]: Use this.axios.

File: frontend/src/services/common/httpService.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    delete(url, params) {
+        return this.axios.delete(url, params);
+    }
+
+    static authRequest() {
+        return authAxios;
+    }
+}
+
+export default HttpService();
```
Comment (nemoipaha) [inline frontend/src/services/common/httpService.js]: new HttpService()

**Pull request merged**